library/src/main/java/com/novoda/downloadmanager/DownloadManager.java

```diff
@@ -224,8 +224,16 @@ public File getDownloadsDir() {
     }
 
     @Override
-    public void addCompletedBatch(CompletedDownloadBatch completedDownloadBatch) {
-        downloader.addCompletedBatch(completedDownloadBatch);
+    public void addCompletedBatch(CompletedDownloadBatch completedDownloadBatch) throws IllegalArgumentException {
```

Comment:
I think this should be synchronous now 🤔 

The `asBatch` can take sometime to create given that it needs to loop through files. It might be useful to return a status too. What do you think?

Reply:
If we decide to make synchronous then I would merge this and I'll add another method to make the refactoring easier. I need this as a base anyway 😄

Reply:
A part from the `asBatch` method (which I don't think is that expensive, what other expensive operations we perform here?